### ISSUE TITLE
Add note about different commands within and outside of app

### ISSUE THREAD

**Issue by cllns** [hanami/cli#51]
We only allow `hanami new` outside of existing Hanami projects, and only allow almost all the other commands inside a Hanami 2 project (`version` is rightfully allowed in both).

For example:
```shell
% hanami version
v2.0.0.beta3
% hanami help
Commands:
  hanami new APP
  hanami version
% hanami new beta_three
% cd beta_three
% hanami help
Commands:
  hanami console                              # App REPL
  hanami generate [SUBCOMMAND]
  hanami install
  hanami middlewares                          # List all the registered middlewares
  hanami routes                               # Inspect application
  hanami server
  hanami version
```

Something like: "There are more commands available when you're inside of an app, but we've detected that you're not in a Hanami 2 folder". We could even list them (on one line, without descriptions) too. And when inside an app something like "`hanami new` is also available when you're outside of an existing Hanami project folder."

I think this could be a "good first issue" for newcomers who'd like to contribute to the Hanami 2 effort.

**Comment by jodosha** [comment]
@cllns
> Something like: ...

What do you mean? To print a warning message?

**Comment by cllns** [comment]
Yes, sorry that wasn't clear. The command behave differently in different context, so we should let the users know which context they're in. Otherwise I expect people may think "Huh? I don't see a `hanami generate` command, it must not have generators" when they first install the gem.

**Comment by jodosha** [comment]
@cllns Sorry for asking the question again: do you want a warning message?
Isn't warning a signal of something wrong that the user did? Why we should warn on default usage?

**Comment by cllns** [comment]
No, not a warning. For example, when outside of an existing Hanami folder:

```bash
% hanami help
Commands:
  hanami new APP
  hanami version

Other commands available within a Hanami app:
  hanami console                              # App REPL
  hanami generate [SUBCOMMAND]
  hanami install
  hanami middlewares                          # List all the registered middlewares
  hanami routes                               # Inspect application
  hanami server
```

**Comment by jodosha** [comment]
This isn't possible.

We register at the runtime two different command registries.

**Comment by cllns** [comment]
Gotcha. What about something generic like

```shell
% hanami help
Commands:
  hanami new APP
  hanami version

Other commands are available within a Hanami project's folder
```
